core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-graph.ts

```diff
@@ -192,6 +192,36 @@ export class WorkflowGraph {
     return Array.from(this.operatorLinkMap.values());
   }
 
+  /**
+   * Returna an array of all input links of an operator in the graph.
+   * @param operatorID
+   */
+  public getInputLinksByOperatorId(operatorID: string): OperatorLink[] {
+    const inputLinks: OperatorLink[] = [];
+    const allLinks: OperatorLink[] = this.getAllLinks();
+    for (let i = 0; i < allLinks.length; i++) {
+      if (isEqual(allLinks[i].target.operatorID, (operatorID))) {
+        inputLinks.push(allLinks[i]);
+      }
+    }
+    return inputLinks;
+  }
+
+  /**
+   * Returna an array of all output links of an operator in the graph.
+   * @param operatorID
+   */
+  public getOutputLinksByOperatorId(operatorID: string): OperatorLink[] {
+    const outputLinks: OperatorLink[] = [];
+    const allLinks: OperatorLink[] = this.getAllLinks();
+    for (let i = 0; i < allLinks.length; i++) {
+      if (isEqual(allLinks[i].source.operatorID, (operatorID))) {
+        outputLinks.push(allLinks[i]);
+      }
+    }
+    return outputLinks;
```

Comment:
Same thing as above, the function can be shorten by using `filter` method provided in javascript.